app/models/user.rb

```diff
@@ -12,29 +13,40 @@ class User < ApplicationRecord
   validates :password, presence: true, allow_nil: true,
    length: {minimum: Settings.user_validates.password_min_length}
 
-  def self.digest string
-    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
-    BCrypt::Password.create string, cost: cost
-  end
+  class << self
+    def digest string
+      cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+      BCrypt::Password.create string, cost: cost
+    end
 
-  def self.new_token
-    SecureRandom.urlsafe_base64
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
   end
 
   def remember
     self.remember_token = User.new_token
     update_attribute :remember_digest, User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
```

Comment:
send là gì thế

Reply:
em có đọc cái này trong metaprogramming, nó dùng để gọi method, ví dụ như:
a.send :puts, "Hello"
cái này em hiểu là nó gọi đến method puts và truyền vào đó "Hello", kết quả sẽ là in ra "Hello"
ngoài ra thì có thể sử dụng string thay cho symbol khi gọi phương thức ("puts" hoặc :puts)